src/main/1A5BC.c

```diff
@@ -892,24 +892,101 @@ INCLUDE_ASM("asm/us/main/nonmatchings/1A5BC", func_8002E698);
 INCLUDE_ASM("asm/us/main/nonmatchings/1A5BC", func_8002E7BC);
 
 // engine_state_3_update_funcs state 3
-INCLUDE_ASM("asm/us/main/nonmatchings/1A5BC", func_8002E8D4);
+void func_8002E8D4(struct EngineObj* arg0)
+{
+    struct MiscObj* obj;
+
+    if (arg0->unk3C->state == 2) {
+        obj = find_free_misc_obj();
+        if (obj != NULL) {
+            obj->base.active = 0x41;
+            obj->base.id = 0xF;
+            obj->base.unk2 = 0;
+        }
+        obj = find_free_misc_obj();
+        if (obj != NULL) {
+            obj->base.active = 0x41;
+            obj->base.id = 0xF;
+            obj->base.unk2 = 1;
+        }
+        obj = find_free_misc_obj();
+        if (obj != NULL) {
+            obj->base.active = 0x41;
+            obj->base.id = 0xC;
+            obj->base.unk2 = 9;
+        }
+        arg0->unk1++;
+    }
+}
 
 // engine_state_3_update_funcs state 4
 INCLUDE_ASM("asm/us/main/nonmatchings/1A5BC", func_8002E994);
 
 // engine_state_3_update_funcs state 5
-INCLUDE_ASM("asm/us/main/nonmatchings/1A5BC", func_8002ED80);
+void func_8002ED80(struct EngineObj* arg0)
+{
+    arg0->unk40 = 0;
+    arg0->unk2 = 0;
+    arg0->unk1++;
+}
 
 // engine_state_3_update_funcs state 6
-INCLUDE_ASM("asm/us/main/nonmatchings/1A5BC", func_8002ED98);
+void func_8002ED98(struct EngineObj* arg0)
+{
+    func_800129F0(8);
+    arg0->unk2 = 0;
+    arg0->unk1++;
+}
 
 // engine_state_3_update_funcs state 7
 INCLUDE_ASM("asm/us/main/nonmatchings/1A5BC", func_8002EDD4);
 
 // engine_state_3_update_funcs state 8 (play video)
 INCLUDE_ASM("asm/us/main/nonmatchings/1A5BC", func_8002EF14);
 
-INCLUDE_ASM("asm/us/main/nonmatchings/1A5BC", func_8002F048);
+void func_8002F048(void)
+{
+    struct MiscObj* misc;
+    struct EffectObj* effect;
+
+    switch (*D_8013B80C) {
+    case 0x0:
+        D_8013B80C++;
+        return;
+    case 0x40:
+        D_8013B80C++;
+        effect = find_free_effect_obj();
+        if (effect != NULL) {
+            effect->active = 1;
+            effect->unk1 = 3;
+            effect->unk2 = 0;
+        }
+        misc = find_free_misc_obj();
+        D_8013B808 = misc;
+        if (misc != NULL) {
+            misc->base.active = 0x41;
+            D_8013B808->base.id = 0xD;
+            D_8013B808->base.unk2 = 1;
+        }
+        misc = find_free_misc_obj();
+        if (misc != NULL) {
+            misc->base.active = 0x41;
+            misc->base.id = 0xC;
+            misc->base.unk2 = 8;
+            misc->ext.ready_text.unk50 = D_8013B808;
+            return;
+        }
+    case 0xFF:
+        return;
+    default:
+        if (*D_8013B80C & 0x40) {
```

Comment:
I suspect that D_8013B80C is also an object, like for example this is probably checking base.active, but I couldn't get it to match since D_8013B80C++ only increments by 1.